pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -229,13 +237,52 @@ func filterNonInactive(orignialList *BlockDeviceList) *BlockDeviceList {
 	return filteredList
 }
 
-func filterClaimedDevices(orignialList *BlockDeviceList) *BlockDeviceList {
+func filterNonPartitions(originalList *BlockDeviceList) *BlockDeviceList {
 	filteredList := &BlockDeviceList{
 		BlockDeviceList: &ndm.BlockDeviceList{},
 		errs:            nil,
 	}
-	for _, device := range orignialList.Items {
-		if device.Status.ClaimState == ndm.BlockDeviceClaimed {
+	for _, device := range originalList.Items {
+		if strings.EqualFold(device.Spec.Partitioned, "No") {
+			filteredList.Items = append(filteredList.Items, device)
+		}
+	}
+	return filteredList
+}
+
+func filterNonFSType(originalList *BlockDeviceList) *BlockDeviceList {
```

Comment:
We are deviating from idiomatic maya practices.
In other words, we should be concentrating on implementing Predicates & making use of these predicates from custom list instance.
```go
type PredicateList []Predicate
func (l PredicateList) all() bool {}
func (l PredicateList) any() bool {}

func IsNonFileSystem() Predicate {...}
func IsSparseDevice() Predicate {...}
func IsNonSparseDevice() Predicate {...}

func (l *BlockDeviceList) Filter (filters ...Predicate) *BlockDeviceList {...}
```

Reply:
The above pattern will be followed once when we start using `pkg/blockdevice/v1alpha2` and `v1alpha2` will fallow the latest pattern. Since we are using `old pattern` in `v1alpha1` we went with same pattern

Reply:
ok